### PR TITLE
Retry docker-compose

### PR DESCRIPTION
**Pull request by rtitle**
See https://github.com/DataBiosphere/leonardo/issues/570

Hoping this helps test flakiness due to timeouts pulling the notebooks image from GCR.

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [ ] I've documented my API changes in Swagger

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/resources/jupyter/init-actions.sh [hunk 1]
```diff
@@ -91,8 +115,9 @@ if [[ "${ROLE}" == 'Master' ]]; then
     log 'Installing Docker Compose...'
 
     # Install docker-compose
-    curl -L https://github.com/docker/compose/releases/download/1.18.0/docker-compose-`uname -s`-`uname -m` -o /usr/local/bin/docker-compose
-    sudo chmod +x /usr/local/bin/docker-compose
+    # https://docs.docker.com/compose/install/#install-compose
+    curl -L "https://github.com/docker/compose/releases/download/1.22.0/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
```
Comment (rtitle) [inline src/main/resources/jupyter/init-actions.sh]: Upgrade to latest version of docker-compose

**Comment by rtitle** [comment]
jenkins retest

File: src/main/resources/jupyter/init-actions.sh [hunk 2]
```diff
@@ -65,7 +89,8 @@ if [[ "${ROLE}" == 'Master' ]]; then
     log 'Installing prerequisites...'
 
     # install Docker
-    export DOCKER_CE_VERSION="17.12.0~ce-0~debian"
+    # https://docs.docker.com/install/linux/docker-ce/debian/
+    export DOCKER_CE_VERSION="18.03.0~ce-0~debian"
```
Comment (rtitle) [inline src/main/resources/jupyter/init-actions.sh]: Upgrade to latest docker too

**Comment by rtitle** [comment]
jenkins retest

**Comment by rtitle** [comment]
jenkins retest

File: src/main/resources/jupyter/init-actions.sh [hunk 3]
```diff
@@ -14,6 +14,30 @@ function update_apt_get() {
   return 1
 }
 
+function run_docker_compose() {
+  for ((i = 0; i < 5; i++)); do
+    docker-compose -f /etc/cluster-docker-compose.yaml config
```
Comment (aweng98) [inline src/main/resources/jupyter/init-actions.sh]: qq, if `config` throws error, retry wouldn't make successful, right?

Comment (rtitle) [reply src/main/resources/jupyter/init-actions.sh]: That's a good point -- shouldn't need to retry `config`

File: src/main/resources/jupyter/init-actions.sh [hunk 4]
```diff
@@ -14,6 +14,30 @@ function update_apt_get() {
...
+    if [ $? -ne 0 ]; then
+      sleep 5
+      continue
+    fi
+    docker-compose -f /etc/cluster-docker-compose.yaml pull
```
Comment (aweng98) [inline src/main/resources/jupyter/init-actions.sh]: `/etc/cluster-docker-compose.yaml` variable is less likely make typos

**Comment by rtitle** [comment]
pause/resume flake, sigh.

jenkins retest

File: src/main/resources/jupyter/init-actions.sh [hunk 5]
```diff
@@ -14,6 +14,27 @@ function update_apt_get() {
   return 1
 }
```
Comment (kyuksel) [inline src/main/resources/jupyter/init-actions.sh]: Can we add a comment mentioning we may be retrying?

File: src/main/resources/jupyter/init-actions.sh [hunk 6]
```diff
@@ -14,6 +14,27 @@ function update_apt_get() {
...
 
+function run_docker_compose() {
+  COMPOSE_FILE=$1
+  docker-compose -f $COMPOSE_FILE config
```
Comment (kyuksel) [inline src/main/resources/jupyter/init-actions.sh]: Can we parameterize the max number of retries and the period in between for easier visibility/configurability?

**Pull request merged**